From 8fbf788c96552d7299868d1ffcc43d1b943aa9e2 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Wed, 26 Nov 2014 11:19:03 +0100
Subject: [PATCH] Disable bluetooth support completely

When the handsfree audio manager is initialized it requires the kernel to support the
bluetooth stack but in some cases it does not and therefor we only enable it conditionally
to not waste startup time with waiting for something which isn't available.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 ofono/Makefile.am | 4 ++++
 ofono/src/main.c  | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/ofono/Makefile.am b/ofono/Makefile.am
index f387f9d..607fe06 100644
--- a/ofono/Makefile.am
+++ b/ofono/Makefile.am
@@ -609,6 +609,10 @@ AM_CFLAGS = @DBUS_CFLAGS@ @GLIB_CFLAGS@ $(builtin_cflags) \
 					-DOFONO_PLUGIN_BUILTIN \
 					-DPLUGINDIR=\""$(build_plugindir)"\"
 
+if BLUETOOTH
+AM_CFLAGS = -DENABLE_BLUETOOTH
+endif
+
 AM_CPPFLAGS = -I$(builddir)/include -I$(builddir)/src -I$(srcdir)/src \
 			-I$(srcdir)/gdbus -I$(srcdir)/gisi -I$(srcdir)/gatchat \
 			-I$(srcdir)/btio -I$(srcdir)/gril
diff --git a/ofono/src/main.c b/ofono/src/main.c
index 73bc7ac..299723d 100644
--- a/ofono/src/main.c
+++ b/ofono/src/main.c
@@ -244,7 +244,9 @@ int main(int argc, char **argv)
 
 	__ofono_manager_init();
 
+#ifdef ENABLE_BLUETOOTH
 	__ofono_handsfree_audio_manager_init();
+#endif
 
 	__ofono_plugin_init(option_plugin, option_noplugin);
 
@@ -255,7 +257,9 @@ int main(int argc, char **argv)
 
 	__ofono_plugin_cleanup();
 
+#ifdef ENABLE_BLUETOOTH
 	__ofono_handsfree_audio_manager_cleanup();
+#endif
 
 	__ofono_manager_cleanup();
 
-- 
1.9.1

